aea/context/base.py

```diff
@@ -60,7 +61,7 @@ def agent_name(self) -> str:
         return self._agent_name
 
     @property
-    def public_key(self) -> str:
+    def public_key(self) -> Dict[str, str]:
```

Comment:
Please call it `public_keys`. Then also create the old `public_key` which we will make the default one, so:
def public_key(self) -> str:
    " Get the default public key"
     return self._public_keys['default']

That way you don't break the interface and are backwards compatible.